controller/inc/fragment_handler.go

```diff
@@ -328,16 +327,34 @@ func (i Inc) LocalOrphanedShardsCheck(jobId int, storeResults bool) ([]string, e
 		if len(resultsOrphanedShards) > 0 {
 			err = i.Db.Create(&resultsOrphanedShards).Error
 			if err != nil {
-				return nil, errors.New("Failed to create ResultsOrphanedShard")
+				return nil, fmt.Errorf("failed to create ResultOrphanedShards %s", err)
 			}
 		}
 
+		// Get the JobProgressOrphanedShard
+		var jobProgressOrphanedShard dbfs.JobProgressOrphanedShard
+		if err = i.Db.Where("job_id = ? AND server_id = ?", jobId, i.ServerName).
+			First(&jobProgressOrphanedShard).Error; err != nil {
+			return nil, fmt.Errorf("failed to get JobProgressOrphanedShard: %s", err)
+		}
+
 		// update JobProgressOrphanedShard
-		err = i.Db.Model(&dbfs.JobProgressOrphanedShard{}).
-			Where("job_id = ? and server_id = ?", jobId, i.ServerName).
-			Update("in_progress", false).Error
+		if len(orphanedShards) > 0 {
+			err = i.Db.Model(&dbfs.JobProgressOrphanedShard{}).
+				Where("job_id = ? and server_id = ?", jobId, i.ServerName).
+				Update("in_progress", false).
+				Update("msg", fmt.Sprintf("%s\nServer %s: Found %v Orphaned Shards",
+					jobProgressOrphanedShard.Msg, i.ServerName,
+					len(orphanedShards))).
+				Error
+		} else {
+			err = i.Db.Model(&dbfs.JobProgressOrphanedShard{}).
+				Where("job_id = ? and server_id = ?", jobId, i.ServerName).
+				Update("in_progress", false).
+				Error
+		}
 		if err != nil {
-			return nil, errors.New("Failed to update JobProgressOrphanedShard")
+			return nil, fmt.Errorf("failed to update JobProgressOrphanedShard: %s", err)
```

Comment:
`%s` -> `%w`, same with the changes below

Reply:
Read it up, understood thanks!